### ISSUE TITLE
Two false positive errors: missing semicolon and \[ escape in regex

### ISSUE THREAD

**Issue by prettydiff** [eslint/eslint#9869]
**Tell us about your environment**

* **ESLint Version:** 4.16.0
* **Node Version:** 8.1.2
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "env": {
        "browser": true,
        "es6": true,
        "node": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "semi": [
            "error",
            "always"
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Sample code in question is very long and dynamically generated, so you can find it here: https://gist.github.com/prettydiff/777bbc96d1ac5c226696de6acad481ba

I just ran ESLint directly against that file.

```bash
eslint js/browser.js
```

**What did you expect to happen?**  I expected there would be no warnings from ESLint

**What actually happened? Please include the actual, raw output from ESLint.**

       1:82   error  Missing semicolon                 semi
       1:170  error  Missing semicolon                 semi
    8289:50   error  Unnecessary escape character: \[  no-useless-escape

This is probably two separate defects.  Here are the respective lines of code:

* 8289 - `if ((((/(>[\w\s:]*)?<(\/|!|#)?[\w\s:-\[]+/).test(sample) === true || (/^(\s*<\?xml)/).test(sample) === true) && ((/^([\s\w]*<)/).test(sample) === true || (/(>[\s\w]*)$/).test(sample) === true)) || ((/^(\s*<s((cript)|(tyle)))/i).test(sample) === true && (/(<\/s((cript)|(tyle))>\s*)$/i).test(sample) === true)) {`

That last line is pretty long, but the warning is tripping on `/(>[\w\s:]*)?<(\/|!|#)?[\w\s:-\[]+/`, more precisely `[\w\s:-\[]`, which is also invalid.

I understand this may not be a failure in ESLint precisely, but rather your parser.  Nonetheless is something the ESLint the project should be aware of.

**Comment by ljharb** [comment]
`1:82` and `1:170` both seem like they're missing a semicolon after `return`. Perhaps the character indicator is off, but you are unarguably missing two semicolons in that line. (Also not sure why you'd ever run a linter on generated code; linters are for human-written code)

`[\w\s:-\[]` should be `[\w\s:-[]` - the second `[` does not need to be escaped, because it's already inside a character class.

**Comment by prettydiff** [comment]
Ah, thanks.

**Comment by not-an-aardvark** [comment]
Note that the two semi errors are both on line 1 (82 and 170 are the column numbers, not the line numbers).

**Comment by prettydiff** [comment]
Yeah, I caught much later.  You guy just responded to this too fast.  I really should be going to bed.
